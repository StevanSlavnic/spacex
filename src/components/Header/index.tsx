import { Link } from "react-router-dom";

export default function Header() {
  return (
    <nav className="bg-gray-800 py-4 mb-4">
      <div className="container mx-auto flex flex-col sm:flex-row justify-between sm:items-center">
        <Link to="/" className="text-white text-2xl font-bold mb-4 sm:mb-0">
          SpaceX Launches
        </Link>

        <div className="space-x-8">
          <Link to="/" className="text-white text-lg">
            Launches
          </Link>
          <Link to="/favorite-launches" className="text-white text-lg">
            Favorite Launches
          </Link>
        </div>
      </div>
    </nav>
  );
}
