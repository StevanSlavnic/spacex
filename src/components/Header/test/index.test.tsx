import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import Header from "../index";

describe("Header component", () => {
  it("renders without crashing", () => {
    const { container } = render(
      <Router>
        <Header />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render the header with correct links", () => {
    const { getByText } = render(
      <Router>
        <Header />
      </Router>
    );

    expect(getByText("SpaceX Launches")).toBeInTheDocument();
    expect(getByText("Launches")).toBeInTheDocument();
    expect(getByText("Favorite Launches")).toBeInTheDocument();
  });

  it("should have correct link paths", () => {
    const { getByText } = render(
      <Router>
        <Header />
      </Router>
    );

    expect(getByText("SpaceX Launches").closest("a")).toHaveAttribute(
      "href",
      "/"
    );
    expect(getByText("Launches").closest("a")).toHaveAttribute("href", "/");
    expect(getByText("Favorite Launches").closest("a")).toHaveAttribute(
      "href",
      "/favorite-launches"
    );
  });
});
