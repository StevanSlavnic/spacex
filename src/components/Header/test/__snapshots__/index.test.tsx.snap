// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders without crashing 1`] = `
<nav
  class="bg-gray-800 py-4 mb-4"
>
  <div
    class="container mx-auto flex flex-col sm:flex-row justify-between sm:items-center"
  >
    <a
      class="text-white text-2xl font-bold mb-4 sm:mb-0"
      href="/"
    >
      SpaceX Launches
    </a>
    <div
      class="space-x-8"
    >
      <a
        class="text-white text-lg"
        href="/"
      >
        Launches
      </a>
      <a
        class="text-white text-lg"
        href="/favorite-launches"
      >
        Favorite Launches
      </a>
    </div>
  </div>
</nav>
`;
